pdl-compiler/scripts/generate_python_backend.py

```diff
@@ -935,7 +935,10 @@ def generate_packet_parser(packet: ast.Declaration) -> List[str]:
         # is given for specialization.
         for _, child in children:
             specialization.append("try:")
-            specialization.append(f"    return {child.id}.parse(fields.copy(), payload)")
+            specialization.append(f"    child, remain = {child.id}.parse(fields.copy(), payload)")
+            specialization.append("    if len(remain) > 0:")
```

Comment:
I find it my pythonic to check for emptiness like this:

```suggestion
            specialization.append(f"    child, remainder = {child.id}.parse(fields.copy(), payload)")
            specialization.append("    if remainder:")
```